### DISCUSSION TITLE
Detailed discussions about animation nodes

### DISCUSSION THREAD

**Discussion by macdroid53** [JacquesLucke/animation_nodes#1850] [Q&A]
I apologize, I realize this is not an issue with animation nodes. This is a use question. I'm looking for the right place to ask such questions.

I have watched a lot of videos and read the docs as I worked on animations.

But, I have not cracked how to make a reciprocating animation. Think a rack and pinion where the rack changes direction as the pinion continues to rotate in the same direction.

I have not found ant videos with similar animation and I've asked in several online forums.

Is there a recommended place for discussion of the use animation nodes?

Any pointers or suggestions would be appreciated.

**Comment by OmarEmaraDev** [comment]
You can use the Q/A section in Github discussions. I converted your issue there. Can you share a reference to what you are trying to do?

**Reply by macdroid53** [reply]
sorry, don't use github that much never knew about discussions.

Thanks. 
I'm on a phone ATM, I can post images tomorrow.

**Comment by macdroid53** [comment]
This image shows what I've got so far. The pinion is the driver and the yoke is supposed to go up when the pinion revolves CW and stop after the last tooth of the pinion disengages. Then the pinion continues CW and engages the other side of the yoke and the yoke should move down to it's other disengage point.But, I haven't figured out how to get the yoke to reverse. (It may be I don't yet know the extent of the tools available in animation nodes. It may also be that the animation could be driven by frames and those control both the pinion and yoke. I'm looking for any input, as I'm new to animation nodes.
![Peek 2022-04-21 08-38](https://user-images.githubusercontent.com/6831752/164747844-b40f68e9-42ba-4b73-aa37-a89b67bfe206.gif)
)

**Comment by OmarEmaraDev** [comment]
The easiest way to do that would be to use a chain of animate nodes. See [here for more information](https://blender.stackexchange.com/questions/135482/sequencing-animations-in-animation-nodes/135583#135583). 

Essentially, we delay the input by 90 degrees (1/2 PI), then animate it from 0 to 1 for 90 degrees, then delay it again for 90 degrees, and finally animate it in reverse from 1 to 0 in 90 degrees. That means it will start at zero and stay zero for 90 degrees, then increase to 1 for the duration of 90 degrees, then stay at 1 for 90 degrees, then finally decrease to 0 in 90 degrees. We also repeat that for 360 degrees (2 PI) to make this animation infinite.

https://user-images.githubusercontent.com/25300994/164799858-8da9f951-f5c5-44e0-88d5-cad867b670bb.mp4

**Reply by macdroid53** [reply]
I believe I've duplicated that but the Result of the second animate does not change.
![Peek 2022-04-23 09-42](https://user-images.githubusercontent.com/6831752/164900739-e83e0663-2d0b-4eaa-ae34-b253f70a390e.gif)

**Reply by OmarEmaraDev** [reply]
Do not write to Blender and read from it again in the same node tree. The rotation of the Alpha object is not guaranteed to be correct here.
Just use the output of the top tree in the bottom one as well.

**Reply by macdroid53** [reply]
Edit: I figured out what I had wrong. I had inverted the value of the top loop to make the rotation CW. Duh. Sorry.
I guess I'm more of a clueless noob than even I think...to me "the output of the top tree in the bottom" means this and I get the same, no change in the Result output of the first _Animate Float_.
![image](https://user-images.githubusercontent.com/6831752/164933704-d364a75d-b474-44a7-ad78-8d7d3a8cdee7.png)
